builtin/credential/userpass/path_users.go

```diff
@@ -95,43 +121,56 @@ func (b *backend) pathUserRead(
 	}, nil
 }
 
-func (b *backend) pathUserWrite(
-	req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
-	name := strings.ToLower(d.Get("name").(string))
-	password := d.Get("password").(string)
-	policies := strings.Split(d.Get("policies").(string), ",")
-	for i, p := range policies {
-		policies[i] = strings.TrimSpace(p)
-	}
-
-	// Generate a hash of the password
-	hash, err := bcrypt.GenerateFromPassword([]byte(password), bcrypt.DefaultCost)
+func (b *backend) userCreateUpdate(req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+	username := strings.ToLower(d.Get("username").(string))
+	userEntry, err := b.user(req.Storage, username)
 	if err != nil {
 		return nil, err
 	}
+	// Due to existence check, user will only be nil if it's a create operation
+	if userEntry == nil {
+		userEntry = &UserEntry{}
+	}
 
-	ttlStr := d.Get("ttl").(string)
-	maxTTLStr := d.Get("max_ttl").(string)
-	ttl, maxTTL, err := b.SanitizeTTL(ttlStr, maxTTLStr)
-	if err != nil {
-		return logical.ErrorResponse(fmt.Sprintf("err: %s", err)), nil
+	// Set/update the values of UserEntry only if fields are supplied
+	if passwordRaw, ok := d.GetOk("password"); ok {
+		// Generate a hash of the password
+		hash, err := bcrypt.GenerateFromPassword([]byte(passwordRaw.(string)), bcrypt.DefaultCost)
+		if err != nil {
+			return nil, err
+		}
+		userEntry.PasswordHash = hash
 	}
 
-	// Store it
-	entry, err := logical.StorageEntryJSON("user/"+name, &UserEntry{
-		PasswordHash: hash,
-		Policies:     policies,
-		TTL:          ttl,
-		MaxTTL:       maxTTL,
-	})
-	if err != nil {
-		return nil, err
+	if policiesRaw, ok := d.GetOk("policies"); ok {
+		policies := strings.Split(policiesRaw.(string), ",")
+		for i, p := range policies {
+			policies[i] = strings.TrimSpace(p)
+		}
+		userEntry.Policies = policies
 	}
-	if err := req.Storage.Put(entry); err != nil {
-		return nil, err
+
+	ttlStrRaw, ttlSet := d.GetOk("ttl")
+	maxTTLStrRaw, maxTTLSet := d.GetOk("max_ttl")
+	if ttlSet || maxTTLSet {
```

Comment:
Good catch. 
Since SanitizeTTL needs to be called if either one of `ttl` or `maxTTL` is set, I have not split this up. Instead, if one of them is set and the other one not, I am using the default value of the field for the one which is not set.
Let me know if you meant something else or if I could do better.